curator-recipes/src/main/java/org/apache/curator/framework/recipes/locks/Reaper.java

```diff
@@ -333,7 +355,20 @@ public void notLeader()
                 reapingIsActive.set(false);
             }
         };
-        localLeaderLatch.addListener(listener);
-        return localLeaderLatch;
+        leaderLatch.addListener(listener);
+
+        reapingIsActive.set(leaderLatch.hasLeadership());
```

Comment:
schedule() gets called on the individual paths in activePaths.  Since this method only gets called during the construction of the Reaper, there won't be any activePaths to schedule. When those paths get added later, they will get scheduled as expected.